lms/static/scripts/frontend_apps/utils/api.ts

```diff
@@ -127,9 +136,19 @@ export async function apiCall<Result = unknown>(
   const resultJSON = await result.json();
 
   if (result.status >= 400 && result.status < 600) {
-    if (result.status === 409 && retryCount < maxRetries) {
+    if (
+      isRetryableResponse(result.status, resultJSON) &&
+      retryCount < maxRetries
+    ) {
       await delay(retryDelay);
-      return apiCall({ ...options, retryCount: retryCount + 1 });
+      return apiCall({
+        ...options,
+        extraHeaders: {
+          ...extraHeaders,
+          ['Retry-Count']: (retryCount + 1).toString(),
```

Comment:
Being pedantic here, but I think the common practice is to prefix custom headers with `X-`, as in `X-Retry-Count`.

Reply:
I did out-pedantic myself before reading this:

https://www.rfc-editor.org/rfc/rfc6648.html#section-3

```
 3.  SHOULD NOT prefix their parameter names with "X-" or similar
       constructs.
```

I reckon the rationale there is that you start naming things X-, then become common usage but they endup stuck with the X-. I don't think there's are risk of that happening here so I don't have a strong opinion.

Reply:
Oh, wow!

I think the rational around adding the `X-` prefix is making sure you don't end up using a header that eventually becomes part of the standard, but I don't think there's a high risk that `Retry-Count` is added to the spec any time soon, without us noticing.

Let's go with `Retry-Count`then.